app/models/locations/location.rb

```diff
@@ -166,20 +166,20 @@ def remove_all_labwares(current_user)
     return if has_child_locations?
 
     # audit that user emptied the location
-    self.create_audit(current_user, Audit::REMOVED_ALL_ACTION)
+    create_audit(current_user, Audit::REMOVED_ALL_ACTION)
 
-    # copy array of labwares to be deleted (labwares will be empty after delete_all)
-    labwares_copy = labwares.each_with_object([]) { |labware, object| object.append(labware) }
-
-    # soft delete labwares to disconnect from relationships including location
-    labwares.delete_all
+    unknown_location = UnknownLocation.get
 
```

Comment:
This code is mostly superseded by the chunk below. I just added a labwares.reset further down to ensure the relationship was updated in memory.